packages/mui-material/src/Avatar/Avatar.js

```diff
@@ -183,7 +185,7 @@ const Avatar = React.forwardRef(function Avatar(inProps, ref) {
         {...imgProps}
       />
     );
-  } else if (childrenProp != null) {
+  } else if (childrenProp != null && (hasAtLeastOneChar || hasNodeChildren)) {
```

Comment:
```suggestion
  } else if (childrenProp != null && childrenProp !== '') {
```

Reply:
If we want to support `false` as well, per https://developer.mozilla.org/en-US/docs/Glossary/Falsy

```suggestion
  } else if (!childrenProp && childrenProp !== 0) {
```

https://github.com/mui/material-ui/blob/7325999b5c62d2ae97b914ffc06c69ecfb53ef27/packages/mui-material/src/Tooltip/Tooltip.js#L506

Reply:
I've reduced the initial overhead and covered both booleans. 

I would appreciate your advice, folks. Right now, one test is invalid because the code has a `propType` that allows it to pass `false` as a child but not `true` as the child, but in both cases boolean will cause an empty render inside `Avatar` instead of the fallback image.
Should I remove the test or extend it so that we expect a console error? If the last is correct, could you reference how similar cases were handled in the codebase?